dace/autodiff/data_forwarding/store.py

```diff
@@ -0,0 +1,664 @@
+
+
+def _get_assign_tasklet(forward_state: SDFGState,
+                        node: nodes.AccessNode,
+                        stored_node: nodes.AccessNode,
+                        last_edge: dgraph.MultiConnectorEdge,
+                        loop_iterators: str,
+                        cuda: bool = False):
+    """
+        """
+    # Create the assign tasklet
+    assign_tasklet_node_in_connector = "in_stored_" + node.data
+    assign_tasklet_node_out_connector = "out_stored_" + node.data
+    assign_tasklet_node = nodes.Tasklet(
+        label=f"__store_{node.data}_assign_",
+        inputs={assign_tasklet_node_in_connector},
+        outputs={assign_tasklet_node_out_connector},
+        code=f"{assign_tasklet_node_out_connector} = {assign_tasklet_node_in_connector}",
+    )
+
+    # Add it to the state
+    forward_state.add_node(assign_tasklet_node)
+
+    # Connect it to the last map entry node
+    # Create the memlet for the assignment
+    # This will be the same as the memlet going to the tasklet
+    assign_memlet_data = copy.deepcopy(last_edge.data)
+    assign_block = assign_tasklet_node
+    assign_block_in_connector = assign_tasklet_node_in_connector
+    return_node = assign_tasklet_node
+    return_connector = assign_tasklet_node_out_connector
+    param_dict = {}
+    memlet_access_iterators = []
+
+    # We check the incoming memlet volumn
+    if assign_memlet_data.volume != 1:
+        # We need to add a map to iterate through the missing dimensions
+        # For this we will create an assign block containing a map
+
+        # First, Get the missing dimensions
+        # Iterate through the subset
+        for element in last_edge.data.subset:
+            if str(element[0]) in last_edge.data.free_symbols:
+                # This is a symbol we will keep in the store memlet
+                memlet_access_iterators.append(str(element[0]))
+            else:
+                # This is a range tuple we need to add an iterator for
+                # Create a random new free symbol
+                free_symbol = "si"
+                i = 0
+                while free_symbol in memlet_access_iterators or free_symbol in forward_state.free_symbols:
+                    free_symbol = f"{free_symbol}_{i}"
+                    i += 1
+                memlet_access_iterators.append(free_symbol)
+                param_dict.update({free_symbol: element})
+
+        # Create the map and add it to the SDFG
+        map = nodes.Map("flatten_assignment_map",
+                        params=list(param_dict.keys()),
+                        ndrange=list(param_dict.values()),
+                        schedule=dtypes.ScheduleType.GPU_Device if cuda else dtypes.ScheduleType.Default)
+        map_entry = nodes.MapEntry(map)
+        map_exit = nodes.MapExit(map)
+        forward_state.add_nodes_from([map_entry, map_exit])
+
+        # Add the necessary connectors
+        assert map_entry.add_in_connector(f"IN_store_block")
```

Comment:
Why don't you use the mapped-tasklet API here? Also, we typically use numbered connectors for data and use names for dynamic map inputs (but I don't think this is enforced).